proteus/NumericalFlux.py

```diff
@@ -807,7 +807,8 @@ def setDirichletValues(self,ebqe):
             for (ebNE,k),g,x in zip(list(self.DOFBoundaryConditionsDictList[ci].keys()),
                                     list(self.DOFBoundaryConditionsDictList[ci].values()),
                                     list(self.DOFBoundaryPointDictList[ci].values())):
-                self.ebqe[('u',ci)][ebNE,k]=g(x,self.vt.timeIntegration.t)
+                self.ebqe[('u',ci)][ebNE,k]=g(x,self.vt.timeIntegration.t, numpy.array(self.vt.ebqe['n'][ebNE,k]))
```

Comment:
Failure here

Reply:
Can you post the stack trace?

Reply:
```
Traceback (most recent call last):
  File "/home/HR/asd/PROTEUS/proteus//centos/bin/parun", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/home/HR/asd/PROTEUS/proteus/scripts/parun", line 601, in <module>
    runName + '_init_prof')
  File "/home/HR/asd/PROTEUS/proteus/proteus/Profiling.py", line 198, in __call__
    return func(*func_args, **func_kwargs)
  File "/home/HR/asd/PROTEUS/proteus/proteus/NumericalSolution.py", line 614, in __init__
    self.allocateModels()
  File "/home/HR/asd/PROTEUS/proteus/proteus/NumericalSolution.py", line 691, in allocateModels
    OneLevelTransportType=p.LevelModelType)
  File "/home/HR/asd/PROTEUS/proteus/proteus/Transport.py", line 6147, in __init__
    PhiSpaceTypeDict=phiSpaces)
  File "/home/HR/asd/PROTEUS/proteus/proteus/Transport.py", line 6374, in initialize
    movingDomain=movingDomain)
  File "/home/HR/asd/PROTEUS/proteus/proteus/mprans/VOF.py", line 866, in __init__
    self.numericalFlux.setDirichletValues(self.ebqe)
  File "/home/HR/asd/PROTEUS/proteus/proteus/NumericalFlux.py", line 810, in setDirichletValues
    self.ebqe[('u',ci)][ebNE,k]=g(x,self.vt.timeIntegration.t, numpy.array(self.vt.ebqe['n'][ebNE,k]))
KeyError: 'n'

```

Reply:
It looks to me like this is failing because VOF doesn't have external boundary normals, and that will be true in general that not all models need them. That's why I used a try/except block to allow specifification of BC's without the normal argument. Can you clarify why you can't use the original try/except approach?

Reply:
With the except blocks it would enter an infinite loop and would put zero values to normals. I tried putting default normals to none, and it would raise an error. I then tried to remove the try command to see where it is failing and then it got me to this

Reply:
Perhaps I should put the try command here

Reply:
@cekees I will try to set ebqe['n] to ebqe_v['n'], hopefully this will do the work